src/core/coap/coap_secure.cpp

```diff
@@ -152,6 +152,35 @@ void CoapSecure::SetSslAuthMode(bool aVerifyPeerCertificate)
 
 #endif // OPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE
 
+#if OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
```

Comment:
Maintaining different methods based on whether block-wise feature is enabled or not ideal. Do you have an idea on the code size impact if we always included the block-wise specific parameters but never used them if the block-wise build feature is not enabled?

Reply:
With block-wise transfer enabled, the code size of the different binaries increase as follows (all values are decimal and in bytes):
ot-cli-ftd:
text: +2736  bss: +214
ot-cli-mtd:
text: +2720  bss: +96
ot-ncp-ftd:
text: +2656  bss: +208
ot-ncp-mtd:
text: +2640  bss: +80

Reply:
Thanks for the information. What I meant is more of a middle ground, where we update the methods to always include the block-wise transfer parameters so that we do not need to maintain two different versions of them. However, the bulk of the implementation we can still feature flag to minimize code/memory. Thoughts?

Reply:
I can't really see what feature would be beneficial to be included all the time, except for the method to add the coap-block-wise option to a coap message which already is. Everthing else i see as routines which depend on each other. But if you have suggestions, i will take a look at them gladly.